app/addons/documents/actions.js

```diff
@@ -0,0 +1,164 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  'app',
+  'api',
+  'addons/documents/resources',
+  'addons/documents/actiontypes'
+],
+function (app, FauxtonAPI, Documents, ActionTypes) {
+  var ActionHelpers = {
+    createNewDesignDoc: function (id, database) {
+      var designDoc = {
+        _id: id,
+        views: {
+        }
+      };
+
+      return new Documents.Doc(designDoc, {database: database});
+    },
+
+    findDesignDoc: function (designDocs, designDocId) {
+      return designDocs.find(function (doc) {
+        return doc.id === designDocId;
+      }).dDocModel();
+
+    }
+  };
+
+  return {
+    //helpers are added here for use in testing actions
+    helpers: ActionHelpers,
+    toggleEditor: function () {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.TOGGLE_EDITOR
+      });
+    },
+
+    selectReduceChanged: function (reduceOption) {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.SELECT_REDUCE_CHANGE,
+        reduceSelectedOption: reduceOption
+      });
+    },
+
+    newDesignDoc: function () {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.NEW_DESIGN_DOC
+      });
+    },
+
+    designDocChange: function (id, newDesignDoc) {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.DESIGN_DOC_CHANGE,
+        newDesignDoc: newDesignDoc,
+        designDocId: id
+      });
+    },
+
+    editIndex: function (options) {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.EDIT_INDEX,
+        options: options
+      });
+    },
+
+    saveView: function (viewInfo, designDocs) {
+      var designDoc;
+
+      if (_.isUndefined(viewInfo.designDocId)) {
+        FauxtonAPI.addNotification({
+          msg:  "Please enter a design doc name.",
+          type: "error",
+          selector: "#define-view .errors-container",
```

Comment:
i just found several usages of `selector`, we are now starting to avoid using error containers for everything that has a submit-button, and start using the global notification. like mentioned yesterday in the standup.